### PR TITLE
Feature/0347 finish unary vectorization

### PR DESCRIPTION
**Pull request by bob-carpenter**
#### Submisison Checklist

- [x] Run unit tests: `./runTests.py test/unit`
- [x] Run cpplint: `make cpplint`
- [x] Declare copyright holder and open-source license: see below

#### Summary:

Adds unary vectorization for remaining unary real-valued functions.

This required adding `double`, `int`, `var`, and `fvar<T>` specializations to many of the functions.  Also the headers got rearranged a bit and some tests were moved to appropriate directories based on their scal/arr/mat typing.

#### Intended Effect:

Allow Stan to have all its unary functions vectorized.

#### How to Verify:

Unit tests.

#### Side Effects:

No.

#### Documentation:

API doc for the new functions.

#### Reviewer Suggestions: 

Anyone.

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting (this will be you or your assignee, such as a university or company):

Columbia University

By submitting this pull request, the copyright holder is agreeing to license the submitted work under the following licenses:
- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
- Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

### CODE REVIEW

File: stan/math/fwd/scal/fun/Phi_approx.hpp [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+#include <cmath>
+
+namespace stan {
+  namespace math {
+
+    /**
+     * Return an approximation of the unit normal cumulative
+     * distribution function (CDF).
+     *
+     * @param x argument
```
Comment (syclik) [inline stan/math/fwd/scal/fun/Phi_approx.hpp]: add `@tparam` to doxygen doc

File: stan/math/fwd/scal/fun/inv_logit.hpp [hunk 2]
```diff
@@ -9,9 +9,7 @@ namespace stan {
   namespace math {
 
     template <typename T>
-    inline
-    fvar<T>
-    inv_logit(const fvar<T>& x) {
+    inline fvar<T> inv_logit(const fvar<T>& x) {
```
Comment (syclik) [inline stan/math/fwd/scal/fun/inv_logit.hpp]: doc for this function

File: stan/math/fwd/scal/fun/log1m_inv_logit.hpp [hunk 3]
```diff
@@ -2,20 +2,26 @@
...
+#include <cmath>
 
 namespace stan {
   namespace math {
 
+    /**
+     * Return the natural logarithm of one minus the inverse logit of
+     * the specified argument.
+     *
+     * @param x argument
```
Comment (syclik) [inline stan/math/fwd/scal/fun/log1m_inv_logit.hpp]: add `@tparam`

File: stan/math/prim/mat/fun/exp2.hpp [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+namespace stan {
+  namespace math {
+
+    /**
+     * Structure to wrap exp2() so it can be vectorized.
+     */
+    struct exp2_fun {
+      /**
+       * Return the inverse hypberbolic cosine of the specified argument.
```
Comment (jgabry) [inline stan/math/prim/mat/fun/exp2.hpp]: Just happened to notice that the comment here looks like it's referring to `acosh` rather than `exp2`.

Comment (syclik) [inline stan/math/prim/mat/fun/exp2.hpp]: I have no idea what's going on with the diff; this is correctly documented on the current branch.

**Comment by syclik** [comment]
@bob-carpenter, I updated the doc. This is good to go as soon as the tests pass.

**Comment by bob-carpenter** [comment]
Thanks!

**Comment by syclik** [comment]
Jenkins, retest this please.

**Pull request merged**